service/storage_service.cc

```diff
@@ -471,29 +471,33 @@ void storage_service::maybe_start_sys_dist_ks() {
     _sys_dist_ks.invoke_on_all(&db::system_distributed_keyspace::start).get();
 }
 
-/* If we're not a seed node or there are seeds other than us,
- * wait until we contact any of them and learn what tokens are used by other nodes in the cluster,
- * with a timeout.
+/* Wait until we retrieve the tokens of all seeds in the cluster, with a timeout.
  *
- * Returns true if we're the only seed or we've managed to learn the other nodes' tokens before timeouting.
+ * Returns true if we've managed to learn all seeds' tokens before timeouting.
  */
 // Run in seastar::async context.
-static bool wait_for_other_tokens(const locator::token_metadata& tm,
-        const std::set<inet_address>& seeds, inet_address my_ep, abort_source& as) {
-    if (seeds.size() == 1 && seeds.count(my_ep)) {
-        return true;
-    }
+static bool wait_for_seed_tokens(const locator::token_metadata& tm,
+        const std::set<inet_address>& seeds, abort_source& as) {
+    std::unordered_set<inet_address> known_endpoints;
+    known_endpoints.reserve(tm.get_token_to_endpoint().size());
```

Comment:
Wouldn't it be better to call tm.get_leaving_endpoints() ?

Reply:
`get_leaving_endpoints()` is completely unrelated to what we're doing here.

Reply:
Sorry, I misread it and thought they are living_endpoinds :D. In general, I was looking for a way to get the set/list of endpoints not a map from token to endpoint.

Reply:
Living endpoints = endpoints which we recently contacted

but here, we don't care whether a node is alive or not. We care whether it's tokens are in the token ring.

Reply:
Sure. That's why I was asking. Question about living endpoints isn't as stupid as about leaving endpoints :D